### PR TITLE
Tests for auto-polygon

### PR DESCRIPTION
**Pull request by martinmcneela**
Ticket: https://edozohq.atlassian.net/browse/MAPS-896

Notes: I have added an extra scenario for when a user clicks and there is not inspire data, will pop that in the ticket too.
﻿

### CODE REVIEW

File: cypress/integration/Autopolygon/Autopolygon.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Background
+ */
+Before(() => {
+  cy.server();
+  cy.route("GET", `${Cypress.env("apiUrl")}/polygon?*`).as("getAutoPolygon");
+});
+
+When(`I go from {string} to {string}`, (freeLayer, paidLayer) => {
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon/Autopolygon.js:11]: This step is duplicated from one in PurchaseMap. Can you move it into the common folder and have them both use the same one please?

File: cypress/integration/Autopolygon.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Feature: Auto-polygon
+
+    Background:
+    Given I open stage
+    And I login with default details
+
+    @has-data
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon.feature:7]: Do these annotations do anything specifically?

File: cypress/integration/Autopolygon.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    Scenario: Add auto-polygon to MasterMap where data exists
+    When I go from 'street map' to 'MasterMap'
+    And I click to select 'autoPolygon' tool
+    And I click the map where data exists
+    Then Add the polygon to the map where data exists
+    
+    @has-not-data
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon.feature:14]:
```suggestion
    @has-no-data
```

File: cypress/integration/Autopolygon.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    Background:
+    Given I open stage
+    And I login with default details
+
+    @has-data
+    Scenario: Add auto-polygon to MasterMap where data exists
+    When I go from 'street map' to 'MasterMap'
+    And I click to select 'autoPolygon' tool
+    And I click the map where data exists
+    Then Add the polygon to the map where data exists
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon.feature:12]: I think this makes more sense being : Then The polygon is added to the map

File: cypress/integration/Autopolygon.feature [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    
+    @has-not-data
+    Scenario: Add auto-polygon to MasterMap where no data exists
+    When I go from 'street map' to 'MasterMap'
+    And I click to select 'autoPolygon' tool
+    And I click the map where data doesn't exist
+    Then Try to add the polygon to the map where no data exists
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon.feature:19]: Similarly I think this one should be: Then The modal is shown and no polygon is added

File: cypress/integration/Autopolygon/Autopolygon.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import { When, And, Then, Before } from "cypress-cucumber-preprocessor/steps";
+
+/**
+ * Background
+ */
+Before(() => {
+  cy.server();
+  cy.route("GET", `${Cypress.env("apiUrl")}/polygon?*`).as("getAutoPolygon");
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon/Autopolygon.js:8]: I don't think you need the Cypress.env part

File: cypress/integration/Autopolygon/Autopolygon.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+});
+
+When(`I go from {string} to {string}`, (freeLayer, paidLayer) => {
+  cy.findByText(new RegExp(freeLayer, "i")).click();
+  cy.findByText(new RegExp(paidLayer, "i")).click();
+  cy.findByRole("button", { name: /ok/i }).click();
+});
+
+And(`I click to select {string} tool`, () => {
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon/Autopolygon.js:17]: Can this be moved to the common steps as well

**Review by zac-robinson — CHANGES_REQUESTED** [review]

File: cypress/integration/Autopolygon.feature (hunk 2 above)
Comment (martinmcneela) [reply cypress/integration/Autopolygon.feature:7]: There is this https://github.com/TheBrainFamily/cypress-cucumber-example#tags-usage, but pretty sure I am not using them correctly so just removed them for now - was trying to make it easy to see what tests were on what scenario.

File: cypress/integration/Autopolygon/Autopolygon.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    expect(xhr.response.body.inspireId, "ID to be a number").to.be.a("number");
+    expect(xhr.response.body.polygon, "Polygon to be a string").to.be.a(
+      "string"
+    );
+  });
+});
+
+/**
+ * has-not-data
+ */
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon/Autopolygon.js:35]: These comments can be removed

File: cypress/integration/Autopolygon/Autopolygon.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Background
+ */
+Before(() => {
+  cy.server();
+  cy.route("GET", `/polygon?*`).as("getAutoPolygon");
+});
+
+When(`I click to select {string} tool`, () => {
```
Comment (zac-robinson) [inline cypress/integration/Autopolygon/Autopolygon.js:11]: I think I missed it last time, can this go into common please. We will need this in lots of places going forward

**Review by zac-robinson — APPROVED** [review]

File: cypress/integration/Autopolygon/Autopolygon.js (hunk 9 above)
Comment (martinmcneela) [reply cypress/integration/Autopolygon/Autopolygon.js:11]: Yeah I was thinking about that but didn't do it as the name of the tool will change and I couldn't get it to pull through the `{string}`

Comment (martinmcneela) [reply cypress/integration/Autopolygon/Autopolygon.js:11]: Scratch that, got it to work but needs a small PR to the maps FE app first.

**Review by zac-robinson — CHANGES_REQUESTED** [review]
The tests failed

**Review by zac-robinson — APPROVED** [review]

**Pull request merged**